fd/diff.go

```diff
@@ -98,7 +108,8 @@ func Derivative(f func(float64) float64, x float64, settings *Settings) float64
 // otherwise a new slice will be allocated and returned. Finite difference
 // kernel and other options are specified by settings. If settings is nil,
 // default settings will be used.
-// Gradient panics if the length of dst and x is not equal.
+// Gradient panics if the length of dst and x is not equal, or if the
```

Comment:
In that case a Hessian approximation directly from f and/or its gradient would be more useful rather than a vector of k-th derivatives. Also having a Jacobian approximation would be good.